src/figuresClasses.ts

```diff
@@ -1,11 +1,109 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  color: string;
+
+  constructor(
+    color: string,
+
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    this.color = color;
+
+    if (this.a <= 0) {
+      throw new Error('`a` can`t be lower or equal zero');
+    }
 
-export class Triangle implements Figure {}
+    if (this.b <= 0) {
+      throw new Error('`b` can`t be lower or equal zero');
+    }
+
+    if (this.c <= 0) {
+      throw new Error('`c` can`t be lower or equal zero');
+    }
+
+    const longest: number = Math.max(this.a, this.b, this.c);
+
+    if (
+      (a === longest && a >= b + c) ||
+      (b === longest && b >= a + c) ||
+      (c === longest && c >= b + a)
+    ) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
```

Comment:
You should use `this.a`, `this.b`, and `this.c` instead of `a`, `b`, and `c` to refer to the class properties within the constructor. This will ensure that the correct values are used when checking if the sides can form a triangle.